Sources/ZipkinRecordingStrategy/Span+ZipkinRepresentation.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Jaeger Client Swift open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Jaeger Client Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Jaeger
+import Tracing
+
+extension JaegerSpan {
+    struct ZipkinRepresentation: Encodable {
+        let id: String
+        let traceID: String
+        let parentID: String?
+        let name: String
+        let timestamp: Int64
+        let duration: Int64
+        let kind: SpanKind
+        let localEndpoint: Endpoint
+        let remoteEndpoint: Endpoint?
+        let tags: [String: String]
+
+        private enum CodingKeys: String, CodingKey {
+            case id, name, timestamp, duration, kind, localEndpoint, tags
+            case traceID = "traceId"
+            case parentID = "parentId"
+        }
+    }
+
+    func zipkinRepresentation(forService serviceName: String) -> ZipkinRepresentation? {
+        guard let traceContext = baggage.traceContext, let endTimestamp = self.endTimestamp else { return nil }
+
+        var tags = [String: String]()
+        self.attributes.forEach { label, attribute in
+            tags[label] = attribute.asString()
+        }
+
+        return ZipkinRepresentation(
+            id: traceContext.parent.parentID,
+            traceID: traceContext.parent.traceID,
+            parentID: self.links.first?.baggage.traceContext?.parent.parentID,
+            name: self.operationName,
+            timestamp: self.startTimestamp.microsSinceEpoch,
+            duration: endTimestamp.microsSinceEpoch - self.startTimestamp.microsSinceEpoch,
+            kind: kind,
+            localEndpoint: .init(serviceName: serviceName),
+            remoteEndpoint: nil,
+            tags: tags
+        )
+    }
+}
+
+extension SpanAttribute {
+    fileprivate func asString() -> String {
+        switch self {
+        case .string(let value):
+            return "\(value)"
+        case .int(let value):
+            return "\(value)"
+        case .double(let value):
+            return "\(value)"
+        case .bool(let value):
+            return "\(value)"
+        case .array(let value):
+            return "[\(value.map { $0.asString() }.joined(separator: ", "))]"
```

Comment:
is this correct for nested strings?

Reply:
Not sure what you mean by nested strings ... `.string()`s stored in `.array()`? 
Those will be reported as `[foo, bar,  baz]`. There's also a unit-test covering this: https://github.com/slashmo/jaeger-client-swift/blob/dd7090af8f19b8670230442e3748804ac5b354f0/Tests/ZipkinRecordingStrategyTests/SpanEncodingTests.swift#L62

Reply:
yeah but what is the string is `"hello, oh no"` -> `[foo, hello, ohno]`

Reply:
is this what zipkin expects and it just is wrong to do that I guess?

Reply:
Ah, didn't think of that. It's kind of hard to say what Zipkin expects. Realistically though, I think this is an edge-case. Let me create an issue for that.

Reply:
yeah, definitely edge case; just good to know what zipkin does if it happens

Reply:
Tracked in #3